packages/cloud_firestore/lib/src/document_reference.dart

```diff
@@ -42,6 +42,17 @@ class DocumentReference {
     );
   }
 
+  Future<DocumentSnapshot> get() {
+    return Firestore.channel.invokeMethod('DocumentReference#get',
+        <String, dynamic>{'path': path}).then((Map<String, dynamic> data) {
+      return new DocumentSnapshot._(
+          data['path'], data['data'], Firestore.instance);
+    }).catchError((PlatformException e) {
```

Comment:
If you're using `await` and wanted to catch errors and handle them, you'd wrap the call to `invokeMethod` in `try { ... } on PlatformException catch (e) { ... }`. But I think in this case since you're not really doing anything with the error, I think you probably want to let the error propagate outwards and make the caller can catch it. It is pretty normal for async methods throw exceptions.

Reply:
used `try { ... } catch (e) { print(e); rethrow }`. Seems unnecessary to catch the exception then rethrow it immediately is there a cleaner way to pass the exception on to the caller.